code/game/gamemodes/objective_items.dm

```diff
@@ -2,23 +2,45 @@
 
 //Contains the target item datums for Steal objectives.
 /datum/objective_item
+	/// How the item is described in the objective
 	var/name = "A silly bike horn! Honk!"
+	/// Typepath of item
 	var/targetitem = /obj/item/bikehorn
-	var/list/valid_containers = list() // Valid containers that the target item can be in.
-	var/difficulty = 9001 //vaguely how hard it is to do this objective
-	var/list/excludefromjob = list() //If you don't want a job to get a certain objective (no captain stealing his own medal, etcetc)
-	var/list/altitems = list() //Items which can serve as an alternative to the objective (darn you blueprints)
+	/// Valid containers that the target item can be in.
+	var/list/valid_containers = list()
+	/// Who CARES if this item goes missing (no stealing unguarded items), often similar but not identical to the next list
+	var/list/item_owner = list()
+	/// Jobs which cannot generate this objective (no stealing your own stuff)
+	var/list/excludefromjob = list()
+	/// List of additional items which also count, for things like blueprints
+	var/list/altitems = list()
+	/// Items to provide to people in order to allow them to acquire the target
 	var/list/special_equipment = list()
+	/// Defines in which contexts the item can be given as an objective
 	var/objective_type = OBJECTIVE_ITEM_TYPE_NORMAL
 	/// Whether this item exists on the station map at the start of a round.
 	var/exists_on_map = FALSE
 
-/datum/objective_item/proc/check_special_completion() //for objectives with special checks (is that slime extract unused? does that intellicard have an ai in it? etcetc)
-	return 1
+/// For objectives with special checks (does that intellicard have an ai in it? etcetc)
+/datum/objective_item/proc/check_special_completion()
+	return TRUE
 
+/// Returns true if the target item exists
 /datum/objective_item/proc/TargetExists()
 	return TRUE
 
+/// Returns true if one of the item's owners exists somewhere
+/datum/objective_item/proc/owner_exists()
+	if (!length(item_owner))
+		return TRUE
+	for (var/mob/player as anything in GLOB.player_list)
+		if (!(player.mind?.assigned_role.title in item_owner))
+			continue
+		if (is_centcom_level(player.z))
+			continue
```

Comment:
Can you reprase this in a manner where the logic is easier to parse? I'd say combine this logic into a single "if this job exists, and they're not on Centcom; then true" branch, rather than the negative gauntlet.